southworks/botframework-solutions
# [TypeScript][Samples/Templates] Update TypeScript ESLint version

## Purpose
_What is the context of this pull request? Why is it being done?_
This pull request fixes the following warning of the [TypeScript ESLint](https://github.com/typescript-eslint/typescript-eslint) library

> WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-estree.
>
> You may find that it works just fine, or you may not.
>
> SUPPORTED TYPESCRIPT VERSIONS: >=3.2.1 <3.6.0
> YOUR TYPESCRIPT VERSION: 3.6.2

## Changes
_Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)_
- Updates TypeScgiript ESLint version from `^1.10.2` to `^2.1.0`.
- Fixes several type casting.
- Disables TSLint rule `prefer-type-cast`.



## Tests
_Is this covered by existing tests or new ones? If no, why not?_
*-*
## Testing Steps
1. Open the generator folder and run `npm install` to install dependencies.
2. Open a terminal and execute `npm link` to link the package of the project locally.
3. From a terminal in different folder execute `yo botbuilder-assistant` or `yo botbuilder-assistant:skill`.
4. The output of the generator should not show the message warning specified in the `Purpose` section.

## Feature Plan
_Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues._
*-*

### Checklist
#### General

#### Bots

#### Deployment Scripts
